### ISSUE TITLE
[QUESTION] Consumer not working with multiple partition from producer?

### ISSUE THREAD

**Issue by vanloctech** [mateusjunges/laravel-kafka#232]
Hello,
I want to run multiple consumers by command and I want each consumer (same topic) to process a certain partition by my producer will randomize and pass it to the partition like the code below.
**Producer code:**
```
$key = Str::uuid()->toString();
$data = serialize(new User());
$i = rand(0, 2);
$message = new Message(
    partition: $i,
    body: $data,
    key: $key
);
$producer = Kafka::publishOn('test-topic-partition')
    ->withConfigOptions([
        'partitioner' => 'random',
        'enable.auto.commit' => true
    ])
    ->withMessage($message);

$producer->send();

return response()->json(['id' => $key, 'partition' => $i]);
```

**Consumer code (function handle() in `consume` command):**
```
public function handle()
{
    $consumer = Kafka::createConsumer(topics: ['test-topic-partition']);
    $consumer->withOption('auto.commit.interval.ms', 100);
    $consumer->withOption('auto.offset.reset', 'earliest');
  
    // Using callback:
    $consumer->withHandler(function(KafkaConsumerMessage $message) {
        // Handle your message here
        $this->info($message->getKey());
        $this->info($message->getPartition());
    });
    $consumer = $consumer->build();
    $consumer->consume();
}
```

When run producer i see many partition with random from 0 to 2 but
Result when run command `php artisan consume` on 3 terminal tab:
- Tab 1: Got results with partition 0. Sample result:
```
5d558371-c29b-4a4f-97fd-a9adb5931558
0
```
- Tab 2: No results.
- Tab 3: No results.

But when I open 3 terminal tabs and run the command, only 1 command tab receives data from partition 0, while the other 2 partitions have no results. I read your source but didn't see anything about partitions in consumers.
And i use single zk, single kafka on docker.

**Comment by mateusjunges** [comment]
Hey @vanloctech currently there is no way to assign partitions to a consumer. I was taking a look on how to implement this and looks quite simple, so I'll ship it with version 2 of this package. You can check the draft PR here: https://github.com/mateusjunges/laravel-kafka/pull/234/files

**Comment by mateusjunges** [comment]
This will be available when I release v2.x, so closing this for now

**Comment by vanloctech** [comment]
@mateusjunges Thank you for your reply, I will try your code.

**Comment by vanloctech** [comment]
Hello @mateusjunges, I tried your code on branch v2.x. But that doesn't seem to be the solution I want. 
I found solution on your code version 1.x.

My solution is that I have to create topics manually from kafka UI or command with a fixed number of partitions first.

For example, if I create a topic named: `test-topic-partition` and the number of partitions is 3, then I give random partition when creating producer, then in 3 terminal consumer tabs work perfectly as I want.

I also re-read some of kafka's documentation and it seems like it's not a good idea to increase the number of partitions even though it's possible.

In short, my solution is to just create a topic with a fixed number of partitions before using it.

Anyway thank you for your response, I think your assign partitions to a consumer logic will be helpful to me too.
